src/Connection_PgSQL.php

```diff
@@ -22,6 +22,11 @@ class Connection_PgSQL extends Connection
     public function lastInsertID($m = null)
     {
         // PostGRE SQL PDO requires sequence name in lastInertID method as parameter
-        return $this->connection()->lastInsertID($m->sequence ?: $m->table . '_' . $m->id_field . '_seq');
+        try {
+            return $this->connection()->lastInsertID($m->sequence ?: $m->table . '_' . $m->id_field . '_seq');
+        } catch (\PDOException $e) {
+            // if no sequence defined (we do not always need it), then silence PDO exception
```

Comment:
This is dangerous as exception may come from broken connection etc., you should definitelly check specific error code here. See https://www.php.net/manual/en/pdo.errorinfo.php - `->errorinfo[2]`.

Reply:
You're probably right here. I will make PR.

Reply:
https://github.com/atk4/dsql/pull/195